### DISCUSSION TITLE
Target Home Store Option

### DISCUSSION THREAD

**Discussion by chrisfrederick** [jef/streetmerchant#1001] [Help]
### Description

This program is so much fun, Thanks to everyone who has contributed!

I am scraping for PS5's (like everyone else) and when I get a hit on Target, my browser pops open with my local home stre which always reads out of stock. My email notifaction sends a sreenshot to another location, I am not even getting a web page with the "temporary issue" Is there a configuration I can do to get the actual location store with the stock to open?

As far as i have been able to tell, I dropped in the URL of the target and item in the target.ts file but liek I said, I can only add the local store.

.env

# ** All configuration variables are optional **
# Read https://github.com/jef/streetmerchant#customization for help on customizing this file
#############################################################################################

ASCII_BANNER=""
ASCII_COLOR=""
AUTO_ADD_TO_CART=""
BROWSER_TRUSTED=""
COUNTRY=""
DESKTOP_NOTIFICATIONS="true"
DISCORD_NOTIFY_GROUP=""
DISCORD_WEB_HOOK=""
EMAIL_PASSWORD="[mypassword]"
EMAIL_TO="[myemail]"
EMAIL_USERNAME="[myusername]"
HEADLESS=""
IN_STOCK_WAIT_TIME="300"
LOG_LEVEL=""
LOW_BANDWIDTH="true"
MAX_PRICE_SERIES_3070=""
MAX_PRICE_SERIES_3080=""
MAX_PRICE_SERIES_3090=""
MAX_PRICE_SERIES_RX6800=""
MAX_PRICE_SERIES_RX6800XT=""
MAX_PRICE_SERIES_RX6900XT=""
MAX_PRICE_SERIES_RYZEN5600=""
MAX_PRICE_SERIES_RYZEN5800=""
MAX_PRICE_SERIES_RYZEN5900=""
MAX_PRICE_SERIES_RYZEN5950=""
MAX_PRICE_CORSAIR_SF=""
MICROCENTER_LOCATION="st-louis-park, duluth"
MQTT_BROKER_ADDRESS=""
MQTT_BROKER_PORT=""
MQTT_CLIENT_ID=""
MQTT_PASSWORD=""
MQTT_QOS=""
MQTT_TOPIC=""
MQTT_USERNAME=""
NVIDIA_ADD_TO_CART_ATTEMPTS=""
NVIDIA_SESSION_TTL=""
OPEN_BROWSER=""
PAGE_BACKOFF_MIN=""
PAGE_BACKOFF_MAX=""
PAGE_SLEEP_MIN=""
PAGE_SLEEP_MAX=""
PAGE_TIMEOUT="0"
PAGERDUTY_INTEGRATION_KEY=""
PAGERDUTY_SEVERITY=""
PHILIPS_HUE_API_KEY=""
PHILIPS_HUE_CLOUD_ACCESS_TOKEN=""
PHILIPS_HUE_CLOUD_CLIENT_ID=""
PHILIPS_HUE_CLOUD_CLIENT_SECRET=""
PHILIPS_HUE_CLOUD_REFRESH_TOKEN=""
PHILIPS_HUE_LAN_BRIDGE_IP=""
PHILIPS_HUE_LIGHT_COLOR=""
PHILIPS_HUE_LIGHT_IDS=""
PHILIPS_HUE_LIGHT_PATTERN=""
PHONE_CARRIER="[mycarrier]"
PHONE_NUMBER="[myphonenumber]"
PLAY_SOUND=""
PROXY_ADDRESS=""
PROXY_PORT=""
PUSHBULLET=""
PUSHOVER_TOKEN=""
PUSHOVER_USER=""
PUSHOVER_PRIORITY=""
SCREENSHOT=""
SHOW_ONLY_BRANDS="sony"
SHOW_ONLY_MODELS="ps5 console, ps5 digital"
SHOW_ONLY_SERIES="sonyps5c, sonyps5de"
SLACK_CHANNEL=""
SLACK_TOKEN=""
SMTP_ADDRESS=""
SMTP_PORT=""
STORES="amazon, bandh, bestbuy, gamestop, microcenter, newegg, playstation,target, walmart"
TELEGRAM_ACCESS_TOKEN=""
TELEGRAM_CHAT_ID=""
TWILIO_ACCOUNT_SID=""
TWILIO_AUTH_TOKEN=""
TWILIO_FROM_NUMBER=""
TWILIO_TO_NUMBER=""
TWITCH_ACCESS_TOKEN=""
TWITCH_CHANNEL=""
TWITCH_CLIENT_ID=""
TWITCH_CLIENT_SECRET=""
TWITCH_REFRESH_TOKEN=""
TWITTER_ACCESS_TOKEN_KEY=""
TWITTER_ACCESS_TOKEN_SECRET=""
TWITTER_CONSUMER_KEY=""
TWITTER_CONSUMER_SECRET=""
TWITTER_TWEET_TAGS=""
USER_AGENT=""
WEB_PORT=""

target.ts

import {Store} from './store';

export const Target: Store = {
	labels: {
		inStock: [
			{
				container: '[data-test="preorderButton"]',
				text: ['Preorder now']
			},
			{
				container: '[data-test="shipItButton"]',
				text: ['Ship it']
			}
		],
		maxPrice: {
			container: '[data-test="product-price"]'
		}
	},
	links: [
		{
			brand: 'sony',
			model: 'ps5 console',
			series: 'sonyps5c',
			url: 'https://www.target.com/p/playstation-5-console/-/A-81114595'
		},
		{
			brand: 'sony',
			model: 'ps5 digital',
			series: 'sonyps5de',
			url:
				'https://www.target.com/p/playstation-5-digital-edition-console/-/A-81114596'
		},
		{
			brand: 'microsoft',
			model: 'xbox series x',
			series: 'xboxsx',
			url: 'https://www.target.com/p/xbox-series-x-console/-/A-80790841'
		},
		{
			brand: 'microsoft',
			model: 'xbox series s',
			series: 'xboxss',
			url: 'https://www.target.com/p/xbox-series-s-console/-/A-80790842'
		}
	],
	name: 'target'
};

**Comment by jef** [comment]
Take a look here: https://jef.codes/streetmerchant/help/general/#adding-a-store

This should get you going.

**Comment by cfbender** [comment]
In case anyone else stumbles upon this and wants to give it a go, this is what I have running currently:

```typescript
export const Target: Store = {
  setupAction: async browser => {
    const page = await browser.newPage();
    await page.goto('https://target.com');
    const cookies = await page.cookies();
    const location = cookies.find(cookie => cookie.name === 'fiatsCookie');
    if (location) {
      logger.info('setting target home store cookie');
      await page.setCookie({
        ...location,
        value: '<PUT TARGET COOKIE HERE>',
      });
    }
  }
...
```

where you replace `<PUT TARGET COOKIE HERE>` with the value of the `fiatsCookie` cookie that target sets when you choose a local store. I'm not _really sure_ if this works, but I guess we will see. I will try to remember to edit this if I know that it either works or doesn't.

**Reply by cfbender** [reply]
This may also fail if the cookie was never set, if you don't get that log maybe try defaulting the `find` to some sensible value for the cookie based on whatever is set in your browser

**Reply by jef** [reply]
Cool find. Glad you were able to utilize the `setupAction` extension.

**Reply by jef** [reply]
Do you have to login to get the cookie or is it for anyone that lands on their homepage?

**Reply by cfbender** [reply]
It should be anyone that lands on the page. Seemed like it was setting the cookie as soon as you selected a home store

**Reply by WeDoSys** [reply]
Well i got this to work on the first loaded page only, after that it overwrites/creates a new cookie based on proxy ip location.

**Reply by Mr-Tech-13** [reply]
How do you find the cookie value?

**Reply by cfbender** [reply]
@Mr-Tech-13 you can find it in the browser console. in firefox it's under the 'Storage' tab
